### PR TITLE
Clean vcache temp files

### PR DESCRIPTION
**Pull request by drostron**
Fixes: #2751 
Fixes: #2740

### CODE REVIEW

**Comment by drostron** [comment]
@jedesah Assigning you since you're familiar with the original vcache PR and we paired on the circular interpreter dependency untanglement.

File: core/src/main/scala/quasar/metastore/MetaStoreAccess.scala [hunk 1]
```diff
@@ -56,9 +56,9 @@ trait MetaStoreAccess {
   def lookupViewCache(path: AFile): ConnectionIO[Option[ViewCache]] =
     (Queries.lookupViewCache(path) ∘ (_.vc)).option
 
-  def updateInsertViewCache(path: AFile, viewCache: ViewCache): ConnectionIO[Unit] =
-    runOneRowUpdateOpt(Queries.updateViewCache(path, viewCache)).isEmpty >>= (_.whenM(
-      runOneRowUpdate(Queries.insertViewCache(path, viewCache))))
+  def insertUpdateViewCache(path: AFile, viewCache: ViewCache): ConnectionIO[Unit] =
```
Comment (jedesah) [inline core/src/main/scala/quasar/metastore/MetaStoreAccess.scala]: Since we are changing the name, can we go with `insertOrUpdateViewCache`?

Also, is this change what fixes the flaky test?

Comment (drostron) [reply core/src/main/scala/quasar/metastore/MetaStoreAccess.scala]: A reasonable name update and yes, this is to address the flaky test.

Comment (drostron) [reply core/src/main/scala/quasar/metastore/MetaStoreAccess.scala]: The change is in a separate commit for future convenience.

**Comment by drostron** [comment]
@jedesah Ready for another look

File: core/src/main/scala/quasar/fs/mount/cache/VCache.scala [hunk 2]
```diff
@@ -75,8 +75,13 @@ object VCache {
     case CompareAndPut(k, expect, v) =>
       injectFT[ConnectionIO, S].apply(lookupViewCache(k)) >>= (vc =>
         (vc ≟ expect).fold(
-          (vc.traverse(deleteVCacheFiles[S]) >>
-            injectFT[ConnectionIO, S].apply(insertOrUpdateViewCache(k, v))).as(true),
+          {
+            vc.foldMap(c =>
+              deleteFiles(
+                ((c.dataFile ≠ v.dataFile) ?? List(c.dataFile)) :::
```
Comment (jedesah) [inline core/src/main/scala/quasar/fs/mount/cache/VCache.scala]: I would add a comment here saying something like: "if the data files are the same in the new view cache as the old one, let's not wipe them away and invalidate the cache". Perhaps I'm on a literate programming fad :smile:  but I'm becoming a huge fan of writing a natural language accompaniment when it helps grok the code faster than just looking at the implementation

Comment (jedesah) [reply core/src/main/scala/quasar/fs/mount/cache/VCache.scala]: And it's especially easy to spot places where it took me longer to understand the code than a sentence would have taken me when reviewing someone else's code so I take the opportunity to point them out! :smile:

File: interface/src/main/scala/quasar/main/ViewCacheRefresh.scala [hunk 3]
```diff
@@ -100,18 +102,20 @@ object ViewCacheRefresh {
...
+  ): Q.transforms.CompExecM[Unit] = {
     val fsQ = new FilesystemQueries[S]
 
     for {
       q       <- EitherT(EitherT(
                    (quasar.resolveImports_[S](view.query, basePath).leftMap(nels(_)).run.run ∘ (_.sequence))
                      .liftM[PhaseResultT]))
-      r       <- fsQ.executeQuery(q, view.vars, basePath, tmpDataPath)
-    } yield r
+      r       <- fsQ.queryResults(q, view.vars, basePath, 0L, none)
```
Comment (jedesah) [inline interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: What's the reason for this change?

File: core/src/main/scala/quasar/fs/mount/cache/VCache.scala (hunk 2 above)
Comment (drostron) [reply core/src/main/scala/quasar/fs/mount/cache/VCache.scala]: Happy to oblige

File: core/src/test/scala/quasar/fs/mount/ViewFileSystemSpec.scala [hunk 4]
```diff
@@ -442,7 +442,7 @@ class ViewFileSystemSpec extends quasar.Qspec with TreeMatchers {
       val f = manage.delete(p).run
 
       viewInterpTrace(vc, f) must_=== ViewInterpResultTrace(
-        Vector.empty,
+        traceInterp(f, Map())._1,
```
Comment (jedesah) [inline core/src/test/scala/quasar/fs/mount/ViewFileSystemSpec.scala]: What's this change for?

File: interface/src/main/scala/quasar/main/ViewCacheRefresh.scala (hunk 3 above)
Comment (drostron) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: `executeQuery` uses the given output file _if possible_. This keeps us from having to detect and handle that case separately.

Comment (jedesah) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: Interesting. Not quite certain what _if possible_ means. Like it won't overwrite an existing file? Anyway, I now understand the change

Comment (drostron) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: The example I ran into was a simple query like `select * from zips` returns `.../zips` instead of the provided output file. Since the query result if equivalent to `.../zips` we don't need to create a separate file for those results. When view caching is involved we need to store the result regardless.

Comment (jedesah) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: Ah, I see. Ya, I think it's cool and all that `executeQuery` tries to be smart that way but it also makes it less useful. Here we are going to be streaming all these cached view results in and out of Quasar for no particularly good reason because of this little bit of "smartness" that is probably not used all that often. We might want to revisit that because this will have a serious performance impact on the common case for view caches

File: core/src/test/scala/quasar/fs/mount/ViewFileSystemSpec.scala (hunk 4 above)
Comment (drostron) [reply core/src/test/scala/quasar/fs/mount/ViewFileSystemSpec.scala]: Per [this change](https://github.com/quasar-analytics/quasar/pull/2783/files/ff6bec3ae5c0e0fd6cf052c84c48d90b7f24108b#diff-9ba21f4161d21d904bf64452dd57b37aR165)

File: interface/src/main/scala/quasar/main/ViewCacheRefresh.scala (hunk 3 above)
Comment (jedesah) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: If we think it's useful, we could add a flag to turn that on and off. I think it would definitely be useful to have a "strict" version of `execute` which does exactly what it's told for cases like this

Comment (drostron) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: A fair concern. I'll look at what's involved in making that change to `executeQuery`.

Comment (wemrysi) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: @drostron Would it be possible to set the view cache's data file to be the path returned from `executeQuery` if it differs from the argument path? Would we then need a way to signal that it shouldn't be automatically deleted?

Comment (drostron) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: @wemrysi For sure, and I originally went that direction but switched due to the absence of a copy term that can be pushed down to a connector. Signaling is possible but doesn't take care of the caching aspect. In this case the cache would be more of a plain view. In fact before this change we were falling back to the associated view anyway albiet with a continually failing vcache refresh. Also, @jedesah and I were discussing the possibility of just removing this special behavior from `execute`. Perhaps you have a sense for why we might preserve it? It seems if you don't want results written you'd just use `evaluate`.

Comment (wemrysi) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: I wasn't sure if what I suggested would be easier than updating `ExecutePlan`, so I just wanted to mention it.

Getting rid of the `ExecutePlan` maybe-it'll-write-maybe-it-won't behavior is fine with me.

Comment (drostron) [reply interface/src/main/scala/quasar/main/ViewCacheRefresh.scala]: Sounds good, I'll roll with updating `ExecutePlan` then.

**Comment by drostron** [comment]
@jedesah Ready for another look
